src/lib.rs

```diff
@@ -193,9 +193,28 @@ impl<T: Float> Complex<T> {
     /// Computes `e^(self)`, where `e` is the base of the natural logarithm.
     #[inline]
     pub fn exp(self) -> Self {
-        // formula: e^(a + bi) = e^a (cos(b) + i*sin(b))
-        // = from_polar(e^a, b)
-        Self::from_polar(self.re.exp(), self.im)
+        // formula: e^(a + bi) = e^a (cos(b) + i*sin(b)) = from_polar(e^a, b)
+
+        // Treat the corner cases +∞, -∞, and NaN
+        let mut im = self.im;
```

Comment:
I think this may be easier to follow if we unpack both parts, and then use them directly throughout.
```suggestion
        let Complex { re, mut im } = self;
```

Reply:
Thanks for the suggestion, I just implemented, tested, and git-pushed it.